homeassistant/components/alexa/smart_home.py

```diff
@@ -154,17 +181,166 @@ def default_display_categories(self):
         """
         raise NotImplementedError
 
-    def capabilities(self):
-        """Return a list of supported capabilities.
+    def get_interface(self, capability):
+        """Return the given _AlexaInterface.
 
-        If the returned list is empty, the entity will not be discovered.
+        Raises _UnsupportedInterface.
+        """
+        pass
+
+    def interfaces(self):
+        """Return a list of supported interfaces.
 
-        You might find _capability() useful.
+        Used for discovery. The list should contain _AlexaInterface instances.
+        If the list is empty, this entity will not be discovered.
         """
         raise NotImplementedError
 
 
-class _GenericCapabilities(_EntityCapabilities):
+class _AlexaInterface(object):
+    def __init__(self, entity):
+        self.entity = entity
+
+    def name(self):
+        """Return the Alexa API name of this interface."""
+        raise NotImplementedError
+
+    @staticmethod
+    def properties_supported():
+        """Return what properties this entity supports."""
+        return []
+
+    @staticmethod
+    def properties_proactively_reported():
+        """Return True if properties asynchronously reported."""
+        return False
+
+    @staticmethod
+    def properties_retrievable():
+        """Return True if properties can be retrieved."""
+        return False
+
+    @staticmethod
+    def get_property(name):
+        """Read and return a property.
+
+        Return value should be a dict, or raise _UnsupportedProperty.
+
+        Properties can also have a timeOfSample and uncertaintyInMilliseconds,
+        but returning those metadata is not yet implemented.
+        """
+        raise _UnsupportedProperty(name)
+
+    @staticmethod
+    def supports_deactivation():
+        """Applicable only to scenes."""
+        return None
+
+    def serialize_discovery(self):
+        """Serialize according to the Discovery API."""
+        result = {
+            'type': 'AlexaInterface',
+            'interface': self.name(),
+            'version': '3',
+            'properties': {
+                'supported': self.properties_supported(),
+                'proactivelyReported': self.properties_proactively_reported(),
+            },
+            # XXX this is incorrect, but the tests assert it
```

Comment:
If it's incorrect, let's remove it. Shouldn't hurt our interaction with Alexa?

Reply:
I see that you have fixed this in #11973 but I don't want the `dev` branch to be in a broken state.

Reply:
It was broken before -- I just didn't notice. I can re-arrange.

Reply:
Oh, in that case it's fine.